deploy/charts/vault-secrets-webhook/README.md.gotmpl

```diff
@@ -0,0 +1,190 @@
+
+```bash
+$ helm upgrade --namespace vswh --install vswh oci://ghcr.io/bank-vaults/helm-charts/vault-secrets-webhook --set-string securityContext.runAsUser="" --create-namespace
+```
+
+### About GKE Private Clusters
+
+When Google configures the control plane for private clusters, they automatically configure VPC peering between your Kubernetes cluster’s network in a separate Google managed project.
+
+The auto-generated rules **only** open ports 10250 and 443 between masters and nodes. This means that to use the webhook component with a GKE private cluster, you must configure an additional firewall rule to allow your masters CIDR to access your webhook pod using the port 8443.
+
+You can read more information on how to add firewall rules for the GKE control plane nodes in the [GKE docs](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#add_firewall_rules).
+
+{{- end -}}
+
+{{- define "after-config" -}}
+### Certificate options
+
+There are the following options for suppling the webhook with CA and TLS certificates.
+
+#### Generate (default)
+
+The default option is to let helm generate the CA and TLS certificates on deploy time.
+
+This will renew the certificates on each deployment.
+
+```
+certificate:
+    generate: true
+```
+
+#### Manually supplied
+
+Another option is to generate everything manually and specify the TLS `crt` and `key` plus the CA `crt` as values.
+These values need to be base64 encoded x509 certificates.
+
+```yaml
+certificate:
+  generate: false
+  server:
+    tls:
+      crt: LS0tLS1...
+      key: LS0tLS1...
+  ca:
+    crt: LS0tLS1...
+```
+
+#### Using cert-manager
+
+If you use cert-manager in your cluster, you can instruct cert-manager to manage everything.
+The following options will let cert-manager generate TLS `certificate` and `key` plus the CA `certificate`.
+
+```yaml
+certificate:
+  generate: false
+  useCertManager: true
+```
+
+{{- end -}}
+
+{{ define "chart.valuesTableHtml" }}
+
+The following table lists the configurable parameters of the Helm chart.
+
+| Parameter | Type | Default | Description |
+| --- | ---- | ------- | ----------- |
+{{- range .Values }}
+| `{{ .Key }}` | {{ .Type }} | {{ .Default }} | {{ if .Description }}{{ .Description }}{{ else }}{{ .AutoDescription }}{{ end }} |
+{{- end }}
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`.
+
+{{ end }}
+
+
+{{- define "chart.baseHead" -}}
+{{ template "chart.header" . }}
+
+{{ template "chart.kubeVersionBadge" . }}
+
+{{ template "chart.description" . }}
+
+{{ template "chart.homepageLine" . }}
+
+{{ template "before-config" . }}
+{{- end -}}
+
+{{- define "chart.base" -}}
+{{ template "chart.baseHead" . }}
+
+{{ template "chart.valuesSectionHtml" . }}
+
+{{ template "after-config" . }}
+{{- end -}}
```

Comment:
Are these definitions necessary here?

Reply:
No need for these templates at all (this enables full helm docs reconfiguration). Replaced with only data injected before and after values table.